js/DataService.js

```diff
@@ -72,16 +81,57 @@ angular.module('PathOfDamage')
           description: "After damage mitigation, modifiers to damage taken are applied. Flat amounts (±X Damage taken from Y, like Astramentis) are applied first, then the sum of all increases/reductions (% increased/reduced X Damage taken, like Fortify, Shock, and Abyssus) and lastly more/less multipliers (% more/less X Damage taken, like Arctic Armour).",
           tables: {
             flat: {
-              totalName: "Flat Increase"
+              name: "Flat Increase",
+              calcTotal: function () {
+                var totals = {};
+                this.values.forEach(function (flat) {
+                  if (flat.enabled && flat.value && flat.types) {
+                    flat.types.forEach(function (type) {
+                      totals[type] = totals[type] + flat.value || flat.value;
+                    });
+                  }
+                });
+                this.totals = totals;
+              },
+              getTotal: function () {
+                return this.totals.physical || 0;
+              }
             },
             increased: {
-              totalName: "Increased Damage Taken",
-              totalMin: -100
+              name: "Increased Damage Taken",
+              calcTotal: function () {
+                var totals = {};
+                this.values.forEach(function (increased) {
+                  if (increased.enabled && increased.value && increased.types) {
+                    increased.types.forEach(function (type) {
+                      totals[type] = Math.max(totals[type] + increased.value || increased.value, -100);
+                    });
+                  }
+                });
+                this.totals = totals;
+              },
+              getTotal: function () {
+                return this.totals.physical || 0;
+              }
             },
             more: {
-              totalName: "More Damage Taken",
-              totalMin: -100,
-              totalCalc: this.multiplicative
+              name: "More Damage Taken",
+              calcTotal: function () {
+                var totals = {};
+                this.values.forEach(function (more) {
+                  if (more.enabled && more.value && more.types) {
+                    more.types.forEach(function (type) {
```

Comment:
Yeah, I stopped mucking around with the function binding because this is going to get refactored using prototypes to have actual inheritence, which will change all of that. I didn't feel like wasting more time on a good interrim solution when I'll be doing that refactor in the next week.